transactions/pendingtxtracker.go

```diff
@@ -530,14 +550,67 @@ func (tm *PendingTxTracker) StoreAndTrackPendingTx(transaction *PendingTransacti
 }
 
 func (tm *PendingTxTracker) addPending(transaction *PendingTransaction) error {
-	insert, err := tm.db.Prepare(`INSERT OR REPLACE INTO pending_transactions
+	var notifyFn func()
+	tx, err := tm.db.Begin()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err == nil {
+			err = tx.Commit()
+			if notifyFn != nil {
+				notifyFn()
+			}
+			return
+		}
+		_ = tx.Rollback()
+	}()
+
+	exists := true
+	var hash eth.Hash
+
+	err = tx.QueryRow(`
+		SELECT hash
+		FROM
+			pending_transactions
+		WHERE
+			network_id = ?
+		AND
+			from_address = ?
+		AND
+			nonce = ?
+		`,
+		transaction.ChainID,
+		transaction.From,
+		transaction.Nonce).
+		Scan(&hash)
+	if err != nil {
+		if err == sql.ErrNoRows {
+			exists = false
+		} else {
+			return err
+		}
+	}
+
+	if exists {
+		notifyFn, err = tm.DeleteBySQLTx(tx, transaction.ChainID, hash)
```

Comment:
Well, this part was deliberately added, to replace (it will be "cancel tx" in UI at some point) the tx in case that tx cannot be processed for any reason (for example if too low fee was set by the user) in that case all txs coming after that get blocked because of this one.